### PR TITLE
Use fmt::formatter specialization for std::reference_wrapper to avoid undefined behavior

### PR DESCRIPTION
**Pull request by phprus**
Fix UB: https://github.com/fmtlib/fmt/pull/4163#issuecomment-2358995143

### CODE REVIEW

File: include/fmt/std.h [hunk 1]
```diff
@@ -692,15 +692,17 @@ template <typename T, typename Char> struct formatter<std::complex<T>, Char> {
...
-}
-
-}  // namespace std
+FMT_EXPORT
+template <typename T, typename Char>
+struct formatter<std::reference_wrapper<T>, Char,
+                 enable_if_t<is_formattable<remove_cvref_t<T>, Char>::value>>
+    : formatter<remove_cvref_t<T>, Char> {
+  template <typename FormatContext>
+  auto format(const std::reference_wrapper<T>& ref, FormatContext& ctx) const
```
Comment (vitaut) [inline include/fmt/std.h:701]: Let's pass `std::reference_wrapper` by value since it should be cheap to copy.

File: test/std-test.cc [hunk 2]
```diff
@@ -399,4 +399,7 @@ TEST(std_test, format_shared_ptr) {
 TEST(std_test, format_reference_wrapper) {
   int num = 35;
   EXPECT_EQ("35", fmt::to_string(std::cref(num)));
+
+  std::atomic<bool> b(true);
+  EXPECT_EQ(fmt::format("{}", std::cref(b)), "true");
```
Comment (vitaut) [inline test/std-test.cc:404]: How is this different from the above test?

**Review by vitaut — COMMENTED** [review]
Thank you!

File: test/std-test.cc (hunk 2 above)
Comment (phprus) [reply test/std-test.cc]: Removed.

File: include/fmt/std.h (hunk 1 above)
Comment (phprus) [reply include/fmt/std.h]: Done

**Comment by vitaut** [comment]
Merged, thanks!

**Pull request merged**